phd-tests/framework/src/test_vm/mod.rs

```diff
@@ -621,35 +690,41 @@ impl TestVm {
     /// [`Self::wait_for_serial_output`] and returns any text that was buffered
     /// to the serial console after the command was sent.
     pub fn run_shell_command(&self, cmd: &str) -> Result<String> {
-        self.send_serial_str(cmd)?;
+        let amended = self.guest_os.amend_shell_command(cmd);
+        let to_send = amended.as_deref().unwrap_or(cmd);
 
         // If the command is multi-line, it won't be echoed literally.
         // instead, it will (probably) have each line begin with an `>`. so,
         // fix that.
-        let mut echo_cmd = cmd.trim_end().replace('\n', "\n> ");
-        echo_cmd.push('\n');
-
-        self.wait_for_serial_output(&echo_cmd, Duration::from_secs(15))
-            .context("Failed to wait for command line")?;
-        let mut out = self
-            .wait_for_serial_output(
-                self.guest_os.get_shell_prompt(),
-                Duration::from_secs(300),
-            )
-            .context("Failed to wait for prompt running shell command")?;
-        out.truncate(out.trim_end().len());
-        Ok(out)
+        let to_send = to_send.trim_end().replace('\n', "\n> ");
+        self.send_serial_str(&to_send)?;
```

Comment:
wait...why does this work? do Unix shells mind if you send them the `> ` character yourself in multiline strings?

Reply:
It doesn't! I butchered this and didn't notice because none of the currently checked-in tests depend on it. Should be fixed now. I checked in the test that I used to verify this (it also works on WS2022), though I could be convinced to omit it as well, on the theory that the test coverage juice may not be worth the test runtime squeeze.

Reply:
ahh, yup, i had forgotten that my hitherto-unmerged tests in #623  were the only tests that actually _use_ a multi line shell command, so i had mistakenly assumed that CI passing meant that it still worked, somehow! i’m glad we caught this!